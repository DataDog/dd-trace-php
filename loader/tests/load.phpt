--TEST--
The loader is able to load ddtrace
--FILE--
<?php

$filter = function ($name) {
    return in_array($name, ['ddtrace', 'ddtrace_injected', 'dd_library_loader']);
};

$exts = array_filter(get_loaded_extensions(false), $filter);
var_dump(array_values($exts));

$exts = array_filter(get_loaded_extensions(true), $filter);
var_dump(array_values($exts));

?>
--EXPECT--
array(1) {
  [0]=>
  string(7) "ddtrace"
}
array(2) {
  [0]=>
  string(17) "dd_library_loader"
  [1]=>
  string(7) "ddtrace"
}
