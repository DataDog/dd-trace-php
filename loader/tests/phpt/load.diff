001+ array(0) {
002+ }
     array(1) {
       [0]=>
003-   string(7) "ddtrace"
004- }
005- array(2) {
006-   [0]=>
       string(17) "dd_library_loader"
008-   [1]=>
009-   string(7) "ddtrace"
     }
