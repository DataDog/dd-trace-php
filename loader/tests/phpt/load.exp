array(1) {
  [0]=>
  string(7) "ddtrace"
}
array(2) {
  [0]=>
  string(17) "dd_library_loader"
  [1]=>
  string(7) "ddtrace"
}