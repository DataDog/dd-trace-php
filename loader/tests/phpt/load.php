<?php

$filter = function ($name) {
    return in_array($name, ['ddtrace', 'ddtrace_injected', 'dd_library_loader']);
};

$exts = array_filter(get_loaded_extensions(false), $filter);
var_dump(array_values($exts));

$exts = array_filter(get_loaded_extensions(true), $filter);
var_dump(array_values($exts));

?>
