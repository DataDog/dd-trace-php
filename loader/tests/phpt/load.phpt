--TEST--
The loader is able to load ddtrace
--SKIPIF--
<?php if (PHP_MAJOR_VERSION <= 5) die("skip: The loader extension is not compatible with PHP 5"); ?>
--FILE--
<?php

$filter = function ($name) {
    return in_array($name, ['ddtrace', 'ddtrace_injected', 'dd_library_loader']);
};

$exts = array_filter(get_loaded_extensions(false), $filter);
var_dump(array_values($exts));

$exts = array_filter(get_loaded_extensions(true), $filter);
var_dump(array_values($exts));

?>
--EXPECT--
array(1) {
  [0]=>
  string(7) "ddtrace"
}
array(2) {
  [0]=>
  string(17) "dd_library_loader"
  [1]=>
  string(7) "ddtrace"
}
