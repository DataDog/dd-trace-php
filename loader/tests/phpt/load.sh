#!/bin/sh

export SHELL='/bin/zsh'
export LSCOLORS='Gxfxcxdxbxegedabagacad'
export USER_ZDOTDIR='/Users/sydney.tung'
export COLORTERM='truecolor'
export LESS='-R'
export XPC_FLAGS='0x0'
export TERM_PROGRAM_VERSION='1.100.3'
export JAVA_24_HOME='/Library/Java/JavaVirtualMachines/zulu-24.jdk/Contents/Home'
export JAVA_HOME='/Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home'
export __CFBundleIdentifier='com.microsoft.VSCode'
export SSH_AUTH_SOCK='/private/tmp/com.apple.launchd.P5HAvCfa72/Listeners'
export MallocNanoZone='0'
export JAVA_17_HOME='/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home'
export DD_LOADER_PACKAGE_PATH='/home/circleci/app/dd-library-php-ssi'
export HOMEBREW_PREFIX='/opt/homebrew'
export PWD='/Users/sydney.tung/dd-trace-php/loader'
export LOGNAME='sydney.tung'
export JAVA_11_HOME='/Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home'
export VSCODE_GIT_ASKPASS_NODE='/private/var/folders/l3/q1x3z27x2sn4zgzt249swjqm0000gp/T/AppTranslocation/E9C84504-3CEC-46BF-8F20-28B83E1D3A9C/d/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin)'
export VSCODE_INJECTION='1'
export COMMAND_MODE='unix2003'
export JAVA_8_HOME='/Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home'
export HOME='/Users/sydney.tung'
export LANG='en_US.UTF-8'
export LS_COLORS='di=1;36:ln=35:so=32:pi=33:ex=31:bd=34;46:cd=34;43:su=30;41:sg=30;46:tw=30;42:ow=30;43'
export VSCODE_PROFILE_INITIALIZED='1'
export TMPDIR='/var/folders/l3/q1x3z27x2sn4zgzt249swjqm0000gp/T/'
export GIT_ASKPASS='/private/var/folders/l3/q1x3z27x2sn4zgzt249swjqm0000gp/T/AppTranslocation/E9C84504-3CEC-46BF-8F20-28B83E1D3A9C/d/Visual Studio Code.app/Contents/Resources/app/extensions/git/dist/askpass.sh'
export CPPFLAGS='-I/opt/homebrew/opt/openjdk@17/include'
export INFOPATH='/opt/homebrew/share/info:/opt/homebrew/share/info:'
export VSCODE_GIT_ASKPASS_EXTRA_ARGS=''
export TERM='xterm-256color'
export ZSH='/Users/sydney.tung/.oh-my-zsh'
export ZDOTDIR='/Users/sydney.tung'
export USER='sydney.tung'
export VSCODE_GIT_IPC_HANDLE='/var/folders/l3/q1x3z27x2sn4zgzt249swjqm0000gp/T/vscode-git-3a930f5b30.sock'
export HOMEBREW_CELLAR='/opt/homebrew/Cellar'
export SHLVL='2'
export PAGER='less'
export HOMEBREW_REPOSITORY='/opt/homebrew'
export XPC_SERVICE_NAME='0'
export VSCODE_GIT_ASKPASS_MAIN='/private/var/folders/l3/q1x3z27x2sn4zgzt249swjqm0000gp/T/AppTranslocation/E9C84504-3CEC-46BF-8F20-28B83E1D3A9C/d/Visual Studio Code.app/Contents/Resources/app/extensions/git/dist/askpass-main.js'
export PATH='/opt/homebrew/opt/openjdk@17/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/usr/local/share/dotnet:~/.dotnet/tools:/usr/local/munki:/opt/homebrew/opt/openjdk@17/bin'
export ORIGINAL_XDG_CURRENT_DESKTOP='undefined'
export JAVA_21_HOME='/Library/Java/JavaVirtualMachines/zulu-21.jdk/Contents/Home'
export OLDPWD='/Users/sydney.tung/dd-trace-php/loader'
export __CF_USER_TEXT_ENCODING='0x1F6:0x0:0x0'
export TERM_PROGRAM='vscode'
export _='/opt/homebrew/bin/php'
export TEMP='/var/folders/l3/q1x3z27x2sn4zgzt249swjqm0000gp/T'
export NO_INTERACTION='1'
export TEST_PHP_EXECUTABLE='/opt/homebrew/bin/php'
export SKIP_ONLINE_TESTS='1'
export TEST_PHP_EXECUTABLE_ESCAPED=''\''/opt/homebrew/bin/php'\'''
export TEST_PHP_CGI_EXECUTABLE='/opt/homebrew/Cellar/php/8.4.10/bin/php-cgi'
export TEST_PHP_CGI_EXECUTABLE_ESCAPED=''\''/opt/homebrew/Cellar/php/8.4.10/bin/php-cgi'\'''
export TEST_PHPDBG_EXECUTABLE='/opt/homebrew/Cellar/php/8.4.10/bin/phpdbg'
export TEST_PHPDBG_EXECUTABLE_ESCAPED=''\''/opt/homebrew/Cellar/php/8.4.10/bin/phpdbg'\'''
export REDIRECT_STATUS='1'
export QUERY_STRING=''
export PATH_TRANSLATED='/Users/sydney.tung/dd-trace-php/loader/tests/phpt/load.php'
export SCRIPT_FILENAME='/Users/sydney.tung/dd-trace-php/loader/tests/phpt/load.php'
export REQUEST_METHOD='GET'
export CONTENT_TYPE=''
export CONTENT_LENGTH=''
export TZ=''
export TEST_PHP_EXTRA_ARGS=' -n  -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=30719" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "opcache.jit_max_root_traces=100000" -d "opcache.jit_max_side_traces=100000" -d "opcache.jit_max_exit_counters=100000" -d "opcache.protect_memory=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/dev/shm/" -d "zend_extension=/Users/sydney.tung/dd-trace-php/loader/modules/dd_library_loader.so" -d "session.auto_start=0" -d "tidy.clean_output=0" -d "zlib.output_compression=Off"'
export HTTP_COOKIE=''

case "$1" in
"gdb")
    gdb --args '/opt/homebrew/bin/php'  -n   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=30719" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "opcache.jit_max_root_traces=100000" -d "opcache.jit_max_side_traces=100000" -d "opcache.jit_max_exit_counters=100000" -d "opcache.protect_memory=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/dev/shm/" -d "zend_extension=/Users/sydney.tung/dd-trace-php/loader/modules/dd_library_loader.so" -d "session.auto_start=0" -d "tidy.clean_output=0" -d "zlib.output_compression=Off" -f "/Users/sydney.tung/dd-trace-php/loader/tests/phpt/load.php"  2>&1
    ;;
"lldb")
    lldb -- '/opt/homebrew/bin/php'  -n   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=30719" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "opcache.jit_max_root_traces=100000" -d "opcache.jit_max_side_traces=100000" -d "opcache.jit_max_exit_counters=100000" -d "opcache.protect_memory=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/dev/shm/" -d "zend_extension=/Users/sydney.tung/dd-trace-php/loader/modules/dd_library_loader.so" -d "session.auto_start=0" -d "tidy.clean_output=0" -d "zlib.output_compression=Off" -f "/Users/sydney.tung/dd-trace-php/loader/tests/phpt/load.php"  2>&1
    ;;
"valgrind")
    USE_ZEND_ALLOC=0 valgrind $2 '/opt/homebrew/bin/php'  -n   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=30719" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "opcache.jit_max_root_traces=100000" -d "opcache.jit_max_side_traces=100000" -d "opcache.jit_max_exit_counters=100000" -d "opcache.protect_memory=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/dev/shm/" -d "zend_extension=/Users/sydney.tung/dd-trace-php/loader/modules/dd_library_loader.so" -d "session.auto_start=0" -d "tidy.clean_output=0" -d "zlib.output_compression=Off" -f "/Users/sydney.tung/dd-trace-php/loader/tests/phpt/load.php"  2>&1
    ;;
"rr")
    rr record $2 '/opt/homebrew/bin/php'  -n   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=30719" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "opcache.jit_max_root_traces=100000" -d "opcache.jit_max_side_traces=100000" -d "opcache.jit_max_exit_counters=100000" -d "opcache.protect_memory=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/dev/shm/" -d "zend_extension=/Users/sydney.tung/dd-trace-php/loader/modules/dd_library_loader.so" -d "session.auto_start=0" -d "tidy.clean_output=0" -d "zlib.output_compression=Off" -f "/Users/sydney.tung/dd-trace-php/loader/tests/phpt/load.php"  2>&1
    ;;
*)
    '/opt/homebrew/bin/php'  -n   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=30719" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "opcache.jit_max_root_traces=100000" -d "opcache.jit_max_side_traces=100000" -d "opcache.jit_max_exit_counters=100000" -d "opcache.protect_memory=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/dev/shm/" -d "zend_extension=/Users/sydney.tung/dd-trace-php/loader/modules/dd_library_loader.so" -d "session.auto_start=0" -d "tidy.clean_output=0" -d "zlib.output_compression=Off" -f "/Users/sydney.tung/dd-trace-php/loader/tests/phpt/load.php"  2>&1
    ;;
esac