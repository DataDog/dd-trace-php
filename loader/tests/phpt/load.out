array(0) {
}
array(1) {
  [0]=>
  string(17) "dd_library_loader"
}